File: core/src/com/colorflow/screen/LoadingScreen.java
package com.colorflow.screen;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.Screen;
import com.badlogic.gdx.graphics.GL20;
import com.badlogic.gdx.graphics.OrthographicCamera;
import com.badlogic.gdx.scenes.scene2d.Stage;
import com.badlogic.gdx.utils.viewport.ScreenViewport;
import com.colorflow.utility.Position;

public class LoadingScreen implements Screen {

    private OrthographicCamera camera;
    private Stage stage;
    private double t = -Math.PI / 2;

    public LoadingScreen() {
        this.camera = new OrthographicCamera();
        this.camera.setToOrtho(false, Position.getWidthScreen(), Position.getHeightScreen());
        this.camera.update();
        this.stage = new Stage(new ScreenViewport(this.camera));
    }

    @Override
    public void show() {
    }

    @Override
    public void render(float delta) {
        t += Math.PI * delta;
        Gdx.gl.glClearColor((float) (Math.sin(t) / 2 + 0.5),
                (float) (Math.sin(t) / 2 + 0.5),
                (float) (Math.sin(t) / 2 + 0.5), 1);
        Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT);
        stage.act(delta);
        stage.draw();
    }

    @Override
    public void resize(int width, int height) {
    }

    @Override
    public void pause() {
    }

    @Override
    public void resume() {
    }

    @Override
    public void hide() {

    }

    @Override
    public void dispose() {
        stage.dispose();
    }
}

File: core/src/com/colorflow/screen/LoadingScreen.kt
package com.colorflow.screen

import com.badlogic.gdx.Gdx
import com.badlogic.gdx.Screen
import com.badlogic.gdx.graphics.GL20
import com.badlogic.gdx.graphics.Texture
import com.badlogic.gdx.graphics.g2d.Sprite
import com.badlogic.gdx.graphics.g2d.SpriteBatch
import com.colorflow.graphic.Position
import com.colorflow.state.GameState
import com.colorflow.state.ScreenType
import com.colorflow.music.Music
import com.colorflow.state.Score
import kotlin.concurrent.thread
import kotlin.math.sin

class LoadingScreen(
        protected val state: GameState,
        protected val music: Music): Screen {

    private var t = -Math.PI / 2
    private val disclamer = Texture(Gdx.files.internal("skin/disclamer.png"))
    private val sprite = Sprite(disclamer)
    private val batch = SpriteBatch()

    override fun render(delta: Float) {
        t += Math.PI * delta
        Gdx.gl.glClearColor(
                (sin(t) / 2 + 0.5).toFloat()/4+0.5f,
                (sin(t) / 2 + 0.5).toFloat()/4+0.5f,
                (sin(t) / 2 + 0.5).toFloat()/4+0.5f, 1f)
        Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT)
        batch.begin()
        sprite.setPosition((Position.widthScreen - disclamer.width)/2, (Position.heightScreen - disclamer.height)/2)
        sprite.draw(batch)
        batch.end()
    }

    override fun show() {
        Gdx.input.inputProcessor = null
        if(state.current_game != null) {
            thread {
                Gdx.app.debug(this::class.java.simpleName, "loading & preparing track '${state.current_game!!.selected_track}'")
                music.prepare(state.current_game!!.selected_track)
                Gdx.app.postRunnable {
                    Gdx.app.debug(this::class.java.simpleName, "score reset")
                    state.current_game!!.score = Score()
                    state.set_screen(ScreenType.PLAY)
                }
            }
        }
    }

    override fun resize(width: Int, height: Int) {}

    override fun pause() {}

    override fun resume() {}

    override fun hide() {}

    override fun dispose() {}
}
